hudi-spark-datasource/hudi-spark/src/test/java/org/apache/hudi/table/upgrade/TestUpgradeDowngrade.java

```diff
@@ -153,6 +156,42 @@ public void testAutoUpgradeDisabled(HoodieTableVersion originalVersion) throws E
     LOG.info("Auto-upgrade disabled test passed for version {}", originalVersion);
   }
 
+  @ParameterizedTest
+  @MethodSource("metadataTableCorruptionTestVersionPairs")
+  public void testMetadataTableUpgradeDowngradeFailure(HoodieTableVersion fromVersion, HoodieTableVersion toVersion) throws Exception {
+    boolean isUpgrade = fromVersion.lesserThan(toVersion);
+    String operation = isUpgrade ? "upgrade" : "downgrade";
+    LOG.info("Testing metadata table failure during {} from version {} to {}", operation, fromVersion, toVersion);
+
+    HoodieTableMetaClient originalMetaClient = loadFixtureTable(fromVersion);
+    assertEquals(fromVersion, originalMetaClient.getTableConfig().getTableVersion(),
+        "Fixture table should be at expected version");
+
+    HoodieWriteConfig cfg = createWriteConfig(originalMetaClient, true);
+
+    String metadataTablePath = HoodieTableMetadata.getMetadataTableBasePath(
+        originalMetaClient.getBasePath().toString());
+    StoragePath metadataHoodiePath = new StoragePath(metadataTablePath, HoodieTableMetaClient.METAFOLDER_NAME);
+    StoragePath propsPath = new StoragePath(metadataHoodiePath, HoodieTableConfig.HOODIE_PROPERTIES_FILE);
+    StoragePath backupPropsPath = new StoragePath(metadataHoodiePath, HoodieTableConfig.HOODIE_PROPERTIES_FILE_BACKUP);
+
+    String corruptedContent = "CORRUPTED_INVALID_CONTENT\n\nTHIS_IS_NOT_VALID_PROPERTIES_FORMAT";
+    try (OutputStream propsOut = originalMetaClient.getStorage().create(propsPath, true);
+         OutputStream backupOut = originalMetaClient.getStorage().create(backupPropsPath, true)) {
+      propsOut.write(corruptedContent.getBytes());
+      backupOut.write(corruptedContent.getBytes());
+    }
+
+    HoodieUpgradeDowngradeException exception = assertThrows(
+        HoodieUpgradeDowngradeException.class,
+        () -> new UpgradeDowngrade(originalMetaClient, cfg, context(), SparkUpgradeDowngradeHelper.getInstance())
+            .run(toVersion, null)
+    );
+
+    LOG.info("Successfully caught expected exception during {} from {} to {}: {}", 
+        operation, fromVersion, toVersion, exception.getMessage());
```

Comment:
there is a way to assert the exception message as well. 
you can find examples in other tests in this repo. 
can we do that.